src/main/webapp/app/exercises/shared/feedback/feedback-item-service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { Feedback } from 'app/entities/feedback.model';
+import { FeedbackItem, FeedbackItemType } from 'app/exercises/shared/result/result-detail.component';
+import { TranslateService } from '@ngx-translate/core';
+import { computeFeedbackPreviewText } from 'app/exercises/shared/feedback/feedback.util';
+
+export interface FeedbackItemService {
+    /**
+     * Creates a feedback item with a category, title and text for each feedback object.
+     * @param feedbacks The list of feedback objects.
+     * @param showTestDetails
+     */
+    createFeedbackItems(feedbacks: Feedback[], showTestDetails: boolean): FeedbackItem[];
+
+    /**
+     * Filters / Summarizes positive test cases for a student and programming exercise result
+     * @param feedbackItems The list of feedback items
+     * @param showTestDetails
+     */
+    filterFeedbackItems(feedbackItems: FeedbackItem[], showTestDetails: boolean): FeedbackItem[];
+
+    /**
+     * Gets positive test cases without detail texts
+     * @param feedbackItems
+     */
+    getPositiveTestCasesWithoutDetailText(feedbackItems: FeedbackItem[]): FeedbackItem[];
+}
+
+@Injectable({ providedIn: 'root' })
+export class FeedbackItemServiceImpl implements FeedbackItemService {
+    constructor(private translateService: TranslateService) {}
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    createFeedbackItems(feedbacks: Feedback[], showTestDetails: boolean): FeedbackItem[] {
+        return feedbacks.map((feedback) => ({
+            type: FeedbackItemType.Feedback,
+            category: this.translateService.instant('artemisApp.result.detail.feedback'),
+            title: feedback.text,
+            text: feedback.detailText,
+            previewText: computeFeedbackPreviewText(feedback.detailText),
+            positive: feedback.positive,
+            credits: feedback.credits,
+        }));
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    getPositiveTestCasesWithoutDetailText(feedbackItems: FeedbackItem[]): FeedbackItem[] {
+        return [];
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    filterFeedbackItems(feedbackItems: FeedbackItem[], showTestDetails: boolean): FeedbackItem[] {
+        return [...feedbackItems];
+    }
```

Comment:
Missing implementations?

Edit: The service for the programming exercise service seems to override those methods. Why are they in the general interface when being specific for programming exercises?

Reply:
This only makes sense with an upcoming stacked PR. The reason why they are here is that mostly due to a lack of or partial application of abstraction.

They will most likely be removed in the follow-up PR